### ISSUE TITLE
Map efficiency: more compact or otherwise better JS code generation

### ISSUE THREAD

**Issue by jteresco** [TravelMapping/Web#275]
This is probably low priority in the current codebase given #268 but there is definitely room for efficiency improvements in the generated JS code that defines the waypoints and connections.  Leaflet seems to let us load maps with many more markers and polylines than Google Maps did and a limiting factor might now be the time taken by https://github.com/TravelMapping/Web/blob/master/lib/waypoints.js.php to generate the JS code, and the time taken to transfer the resulting code.

Possibilities include shorter variable names to generate smaller JS code text and statically allocated arrays to make them more efficient to load up.  The code we have now is based on early CHM techniques and our data has grown significantly since that was first implemented.

**Comment by yakra** [comment]
What are your thoughts on statically allocated arrays specifically?

**Comment by jteresco** [comment]
I suspect statically allocated arrays will be created more quickly and efficiently.  If nothing else, they know how big they are right away.

**Comment by yakra** [comment]
Still not fully wrapping my head around it.
I think primarily in terms of C++, where the `sizeof` the elements in the array is known beforehand.
In Javascript, OTOH, weak typing *(am I using the term correctly?)* means that each element can be a different size.
Would the difference lie in declaring the whole array in one fell swoop?
For example, view-source:http://yakra.teresco.org/tmtools_demos/tmg2html/DC-region.tmg.html...
Right now:
```Javascript
var vertex = [];
vertex[0] = {lat:38.917917,lon:-76.941805,label:'US50@DC/MD'}
vertex[1] = {lat:38.902397,lon:-76.943741,label:'DC295@NHBurAve'}
// ...
vertex[92] = {lat:38.792435,lon:-77.037388,label:'I-95/I-495/I-495@DC/MD'}
var edge = [];
edge[0] = {begin:86,end:87,qty:1,label:'ClaBarPkwy',shape:[]}
edge[1] = {begin:76,end:77,qty:1,label:'US29',shape:[]}
// ...
edge[87] = {begin:12,end:0,qty:1,label:'US50',shape:[]}
```
Would this be better off as:
```Javascript
var vertex = [
{lat:38.917917,lon:-76.941805,label:'US50@DC/MD'}.
{lat:38.902397,lon:-76.943741,label:'DC295@NHBurAve'},
// ...
{lat:38.792435,lon:-77.037388,label:'I-95/I-495/I-495@DC/MD'}];
var edge = [
{begin:86,end:87,qty:1,label:'ClaBarPkwy',shape:[]},
{begin:76,end:77,qty:1,label:'US29',shape:[]},
// ...
{begin:12,end:0,qty:1,label:'US50',shape:[]}];
```

**Comment by jteresco** [comment]
Yes, I am suspecting that the latter will be more efficient, since JS can know how much memory to allocate right away, rather than resizing over and over as more array elements are added.

I hope to experiment with this soon.  I don't think it will be hard to make the switch.
